L66-core-cancellation-status.md

```diff
@@ -0,0 +1,62 @@
+
+## Background
+
+gRPC core versions up to 10.X specify that the out argument of the `GRPC_OP_RECV_CLOSE_ON_SERVER` operation is a pointer to an int that specifies whether the RPC was cancelled. The definition specified in the comment is
+
+```
+      /** out argument, set to 1 if the call failed in any way (seen as a
+          cancellation on the server), or 0 if the call succeeded */
+```
+
+The phrase "in any way" is not clear, and the broadest (and current) definition is that it accounts for failures caused by issues such as client-side or server-side cancellations, deadlines being exceeded, connections exceeding their maximum age, network resets, *and explicit sending of non-OK status*.
+
+
+### Related Proposals
+
+N/A
+
+## Proposal
+
+Clarify that RPCs will only be marked cancelled if they failed for a reason other than completion with an explicit non-OK status provided using the `GRPC_OP_SEND_STATUS_FROM_SERVER` operation. In practice, this means that RPCs to be considered cancelled are those for which the server was not able to successfully send any kind of status to the client (because the RPC was explicitly cancelled, the deadline was exceeded, because some HTTP configuration parameter like maximum connection age took effect, etc.).
+
+## Rationale
+
+Calling an RPC cancelled if it completes for entirely expected reasons is confusing. Generally speaking, this distinction has not been noticed at the API layers but will become more common once the C++ callback API becomes formalized (since that has an OnCancel reaction for server RPCs). As a result, this issue should be clarified and fixed before the C++ callback API becomes common.
+
+## Implementation
+
+The implementation is contained entirely within a single pull request at https://github.com/grpc/grpc/pull/22991 . This pull request does the following:
+
+1. Decide if a server call is canceled by whether or not it could successfully send status
+1. Fix core tests (wrapped language tests already had the proper expectations)
+1. Mark a call failed according to channelz if it was canceled or if the server sent a non-OK status
+1. Add a C++ test to validate that OnCancel is not called on non-OK explicit status
+1. Fix the core surface comment about this topic and bump the core major API version
+1. Add a C++ comment clarifying the meaning of `IsCancelled`
+
+## Open issues (if applicable)
+
+* Does this affect the core wrapped language APIs?
+  - This distinction is not visible to Ruby or the C++ synchronous API at all since the RPC in both of those cases is complete when returning status. Thus cancellation checks just see the effect of other forms of cancellation.
+  - For the C++ asynchronous API, `IsCancelled` currently doesn't define its result. This change also includes an explciit definition of cancellation for C++.
+  - Python does not expose this functionality to the server application.
+  - For C#, ???
```

Comment:
Please consult with @jtattermusch about this.  We should make sure this entry is filled in before merging this or the implementation PR.

Reply:
I've reached out again to him. I'll let you know when I fill in this content.